src/plugins/reticle/reticle.main.ts

```diff
@@ -0,0 +1,71 @@
+import { IMainThreadContext } from "../../engine/MainThread";
+import { defineModule, getModule, registerMessageHandler } from "../../engine/module/module.common";
+import { ReticleFocusMessage, ReticleFocusMessageType } from "./reticle.common";
+
+interface ReticleModuleState {
+  reticleElement: HTMLElement;
+  parentElement: HTMLElement;
+}
+
+const reticleCss = `
+  position: absolute;
+  top: 50vh;
+  left: 50vw;
+  background: white;
+  width: 6px;
+  height: 6px;
+  border-radius: 10px;
+  border-style: solid;
+  border-width: 1px;
+  margin-top: -3px;
+  margin-left: -3px;
+  z-index: 999;
+  transition: all 0.05s linear;
+`;
+
+const reticleFocusedCss = `
+  ${reticleCss}
+  width: 8px;
+  height: 8px;
+  margin-top: -4px;
+  margin-left: -4px;
+  background: cornflowerblue;
+`;
+
+export const ReticleModule = defineModule<IMainThreadContext, ReticleModuleState>({
```

Comment:
This should probably use a React component instead and just use the `registerMessageHandler` method inside a hook to call `setState` and then use that to pick a class.

Reply:
i considered this but do we really want to involve the vdom for such a tiny feature? i suppose yes if the reticle interfaces with the UI in a more involved way, but couldn't think of how.

Reply:
Yeah, it's gonna be more maintainable for this, perf isn't an issue at all. The best example is the `Stats` or `Editor` components. They just get the `mainThread` reference and then do the same thing you're doing here.

Reply:
Maybe something like this?

```typescript
export function Reticule() {
  const mainThread = useMainThreadContext();
  const [focused, setFocused] = useState<boolean>(false);

  useEffect(() => {
    const onReticleFocus = (ctx: IMainThreadContext, message: ReticleFocusMessageType) => {
      setFocused(message.focused);
    };

    return registerMessageHandler(ctx, ReticleFocusMessage, onReticleFocus);
  }, [mainThread]);

  return <div className={classNames("Reticule", { "Reticule--focused": focused })} />;
}
```